x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/CachingUsernamePasswordRealm.java

```diff
@@ -157,18 +162,36 @@ private void authenticateWithCache(UsernamePasswordToken token, ActionListener<A
                     } else if (credsMatch) {
                         // not authenticated but instead of hammering reuse the result. a new
                         // request will trigger a retried auth
+                        logger.trace(
+                            "realm [{}], provided credentials for user [{}] are invalid (cached result) status:[{}] message:[{}]",
+                            name(),
+                            token.principal(),
+                            cachedResult.authenticationResult.getStatus(),
+                            cachedResult.authenticationResult.getMessage()
+                        );
                         listener.onResponse(cachedResult.authenticationResult);
                     } else {
+                        logger.trace(
+                            "realm [{}], provided credentials for user [{}] do not match (possibly invalid) cached credentials," +
+                                " invalidating cache and retrying",
+                            name(),
+                            token.principal()
+                        );
                         cache.invalidate(token.principal(), listenableCacheEntry);
                         authenticateWithCache(token, listener);
                     }
                 }, listener::onFailure), threadPool.executor(ThreadPool.Names.GENERIC), threadPool.getThreadContext());
             } else {
+                logger.trace(
+                    "realm [{}] does not have a cached result for user [{}]; attempting fresh authentication", name(), token.principal());
                 // attempt authentication against the authentication source
                 doAuthenticate(token, ActionListener.wrap(authResult -> {
                     if (authResult.isAuthenticated() == false || authResult.getUser().enabled() == false) {
+                        logger.trace("realm [{}] did not authenticate user [{}] ([{}])", name(), token.principal(), authResult);
```

Comment:
Would it make sense to differentiate between failed password authentication and user being disabled in this log ?

Reply:
Also should we make this debug ?

Reply:
I think it's worth splitting so that a disabled user can be DEBUG.
Not authenticated definitely shouldn't be DEBUG - it will happen any time a realm doesn't authenticate a user, including in the reserved realm.

Reply:
> Not authenticated definitely shouldn't be DEBUG - it will happen any time a realm doesn't authenticate a user, including in the reserved realm.

+1, makes sense, thanks